### ISSUE TITLE
Installing servers with bonding enabled DURING installation does not work

### ISSUE THREAD

**Issue by jovandeginste** [theforeman/community-templates#401]
We started testing with booting servers with bonding straight after PXE. This has an annoying side effect: the two slave devices have the same mac address :-)

The current way that Foreman gets a device's OS-determined name ([here](https://github.com/theforeman/community-templates/blob/develop/provisioning_templates/snippet/kickstart_ifcfg_get_identifier_names.erb#L10)) does not work then:

```
ip -o link | awk '/#{@interface.inheriting_mac}/ {print $2;}' | sed s/:$//
```

We switched to the following line instead, which seems to work in all our cases (including non-bond servers):

```
grep -l #{@interface.inheriting_mac} /sys/class/net/*/{bonding_slave/perm_hwaddr,address} \
    | awk -F '/' '// {print $5}' | head -1
```

In short: we prioritize bonding slaves (like eth0) over other devices (like bond0).

Anyone care to weigh in on this?

**Comment by jovandeginste** [comment]
+@ziporah

**Comment by ekohl** [comment]
@lzap maybe you have some knowledge here.

**Comment by lzap** [comment]
Hey, this looks good to me. This is only used in Red Hat kickstarts, so it's Linux only, therefore it's fine to access `/sys` directly. Unless @ares has different opinion, please go ahead and make a PR! Thanks

**Comment by mmoll** [comment]
@timogoebel iirc you're also having experiences with this topic...

**Comment by ares** [comment]
I think that makes sense, the only question is whether that works with vlans attached to bond, e.g. devices like `bond0.1`. But that should be fixed soon by http://projects.theforeman.org/issues/13092 so :+1 from me.

**Comment by ziporah** [comment]
A quick verification of your suggestion shows everything still works even if a vlan is tagged using bond0.<vlan id>

```bash
grep -l 14:02:ec:00:79:87 /sys/class/net/*/{bonding_slave/perm_hwaddr,address}     | awk -F '/' '// {print $5}' 
eth0
bond0.200
bond0
eth0
eth1
vlan326
```
And second interface
```bash
grep -l 14:02:ec:00:79:88 /sys/class/net/*/{bonding_slave/perm_hwaddr,address}     | awk -F '/' '// {print $5}'
eth1
````

Normally the 
```bash
|head -1
```
 is added, so only the top interface (this is the right one) is used.

**Comment by lzap** [comment]
Great please go ahead and make a PR, if you do it quick you might be able to hit 1.16 merging window :-)

**Comment by timogoebel** [comment]
@jovandeginste / @ziporah : This change does make a lot of sense. 👍  from me.

**Comment by jovandeginste** [comment]
@timogoebel @lzap @ares @ziporah Made a PR #408
